File: purchase/frontend/src/app/core/services/purchase-order.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { PurchaseOrderDTO } from '../models/purchase-order.model';

@Injectable({
  providedIn: 'root'
})
export class PurchaseOrderService {
  private apiUrl = `${environment.apiUrl}/purchase-orders`;

  constructor(private http: HttpClient) { }

  getAllPurchaseOrders(): Observable<PurchaseOrderDTO[]> {
    return this.http.get<PurchaseOrderDTO[]>(this.apiUrl);
  }

  getPurchaseOrderById(id: number): Observable<PurchaseOrderDTO> {
    return this.http.get<PurchaseOrderDTO>(`${this.apiUrl}/${id}`);
  }

  createPurchaseOrder(purchaseOrder: PurchaseOrderDTO): Observable<PurchaseOrderDTO> {
    return this.http.post<PurchaseOrderDTO>(this.apiUrl, purchaseOrder);
  }

  updatePurchaseOrder(id: number, purchaseOrder: PurchaseOrderDTO): Observable<PurchaseOrderDTO> {
    return this.http.put<PurchaseOrderDTO>(`${this.apiUrl}/${id}`, purchaseOrder);
  }

  deletePurchaseOrder(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}

File: purchase/backend/src/main/java/com/example/purchase/purchaseorder/PurchaseOrderService.java
package com.example.purchase.purchaseorder;

import java.util.Date;
import java.util.List;

public interface PurchaseOrderService {

    // Get all purchase orders
    List<PurchaseOrderDTO> getAllPurchaseOrders();

    // Get purchase order by id
    PurchaseOrderDTO getPurchaseOrderById(Integer id);

    // Get purchase orders by status
    List<PurchaseOrderDTO> getPurchaseOrdersByStatus(String status);

    // Get purchase orders by vendor
    List<PurchaseOrderDTO> getPurchaseOrdersByVendor(Integer vendorid);

    // Get purchase orders by event
    List<PurchaseOrderDTO> getPurchaseOrdersByEvent(Integer eventid);

    // Get purchase orders by cdsid
    List<PurchaseOrderDTO> getPurchaseOrdersByCdsid(String cdsid);

    // Get purchase orders by year
    List<PurchaseOrderDTO> getPurchaseOrdersByYear(int year);

    // Get purchase orders by date range
    List<PurchaseOrderDTO> getPurchaseOrdersByDateRange(Date startDate, Date endDate);

    // Get completed purchase orders
    List<PurchaseOrderDTO> getCompletedPurchaseOrders();

    // Get total order amount by vendor
    Double getTotalOrderAmountByVendor(Integer vendorid);

    // Create purchase order
    PurchaseOrderDTO createPurchaseOrder(PurchaseOrderDTO dto);

    // Update purchase order
    PurchaseOrderDTO updatePurchaseOrder(Integer id, PurchaseOrderDTO dto);

    // Complete purchase order
    PurchaseOrderDTO completePurchaseOrder(Integer id);

    // Reject purchase order
    PurchaseOrderDTO rejectPurchaseOrder(Integer id);

    // Delete purchase order
    void deletePurchaseOrder(Integer id);
}
